react/components/NestedExtensionPoints.tsx

```diff
@@ -27,21 +28,33 @@ export default class NestedExtensionPoints extends PureComponent<Props> {
   }
 
   public render() {
-    const {page, query} = this.props
+    const {page, query, breakPoint} = this.props
     const segments = page.split('/')
     const reverse = segments.slice().reverse()
     // Nest extension points for nested pages
     // a/b/c should render three extension points
     // <a><b><c></c></b></a>
     const getNestedExtensionPoints = (runtime: RenderContext) => {
-      return reverse.reduce((acc: JSX.Element | null, value: string, index: number) => (
-        <ExtensionPoint
-          id={value}
-          query={query}
-          params={this.getPageParams(runtime, segments.slice(0, segments.length - index).join('/'))}>
-          {acc}
-        </ExtensionPoint>
-      ), null as JSX.Element | null)
+      return reverse.reduce((acc: JSX.Element | null, value: string, index: number) => {
+        if (breakPoint && breakPoint.point === value && breakPoint.isLoading) {
```

Comment:
I see there is a desirable behavior that in case of nested pages we should be able to render every "page" until we got one that requires authentication. Like rendering `store` header and footer components but not `store/account` component.

Then I think a better approach would be to have the `<AuthWrapper>` here, and not in `RenderProvider`

```js
return reverse.reduce((acc: JSX.Element | null, value: string, index: number) => (
  <AuthWrapper pages={runtime.pages} page={page} navigate={runtime.navigate}>
    <ExtensionPoint
      id={value}
      query={query}
      params={this.getPageParams(runtime, segments.slice(0, segments.length - index).join('/'))}>
      {acc}
    </ExtensionPoint>
  </AuthWrapper>
), null as JSX.Element | null)
```

Also we could rename `AuthWrapper` to `MaybeAuth`.